.github/workflows/test.yml

```diff
@@ -30,11 +30,14 @@ jobs:
           enable-cache: true
           python-version: ${{ matrix.python-version }}
           version: ${{ env.UV_VERSION }}
-      - name: Install dependencies
-        run: uv sync --group plugin-test --frozen
-      - name: Test with pytest
-        run: |
-          uv run pytest -n logical --dist loadgroup --junit-xml=test-results.xml
+      - name: Install mandatory dependencies
+        run: uv sync --frozen
+      - name: Run tests not requiring optional dependencies
+        run: uv run pytest -m 'not require_optional_deps' -n logical --dist loadgroup --junit-xml test-results.xml
+      - name: Install optional dependencies
+        run: uv sync --frozen --group all --group plugin-test
```

Comment:
Nice. This is getting pretty close to what I had in mind, the only other thing I was thinking is that we take out the `--group all` here. Then we'd just add telegram to the plugin-test group. (Unless there is some advantage I'm not seeing to installing all the extras.) We could add it as `{ include-group = "telegram" }` (like is done for the 'all' group) to avoid listing the version requirements separately again.

Reply:
I've made the changes according to your suggestion.